core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -1289,6 +1292,11 @@ private void checkExponentialCnf(int n) {
     checkSimplify(coalesce(iRef, literal1), "COALESCE(?0.i, 1)");
     checkSimplify(coalesce(iRef, plus(iRef, hRef), literal1, hRef),
         "COALESCE(?0.i, +(?0.i, ?0.h), 1)");
+
+    // not(x) is null should not optimized to x is not null
+    checkSimplify(isNull(not(fRef)), "IS NULL(NOT(?0.f))");
```

Comment:
Please rebase and use `isNull(not(vBool()))`

Reply:
please add case for isNull(not(vBoolNotNull())) as well

Reply:
Thanks! I have modified.